refactor: improve github actions cache optimization

sublimator opened the pull request:
- move ccache configuration logic to dedicated action
- rename conanfile-changed to should-save-conan-cache for clarity
- add daily cron schedule to maintain fresh caches
- implement two-tier ccache system for feature branches
- only save conan cache when conanfile changes on feature branches

sublimator committed:
refactor: improve github actions cache optimization

- move ccache configuration logic to dedicated action
- rename conanfile-changed to should-save-conan-cache for clarity
- add daily cron schedule to maintain fresh caches
- implement two-tier ccache system for feature branches
- only save conan cache when conanfile changes on feature branches

sublimator committed:
fix: remove overly broad ccache restore-key without compiler discriminant

tequdev commented on .github/workflows/xahau-ga-macos.yml:
```diff
@@ -76,6 +76,9 @@ jobs:
           hash_dir: true
           compiler_check: content
+          is_main_branch: ${{ github.ref_name == env.MAIN_BRANCH_NAME }}
+          main_cache_dir: ~/.ccache-main
+          current_cache_dir: ~/.ccache-current
```
Isn't it unnecessary to specify this because it is the same as the default value?

sublimator replied:
Yes, good point, updated

tequdev commented on .github/workflows/xahau-ga-nix.yml:
```diff
@@ -49,6 +49,9 @@ jobs:
           hash_dir: true
           compiler_check: content
+          is_main_branch: ${{ github.ref_name == env.MAIN_BRANCH_NAME }}
+          main_cache_dir: ~/.ccache-main
+          current_cache_dir: ~/.ccache-current
```
same here

tequdev commented on .github/actions/xahau-configure-ccache/action.yml:
```diff
@@ -14,18 +14,53 @@ inputs:
+          echo "CCACHE_DIR=${{ inputs.main_cache_dir }}" >> $GITHUB_ENV
+          
+          # Configure settings for main cache
+          export CONF_PATH="${{ inputs.main_cache_dir }}/ccache.conf"
```
Is it necessary to change the location of the conf file?
You only need to change the cache path and secondary path referenced in the same conf file.

tequdev commented on .github/actions/xahau-configure-ccache/action.yml:
```diff
@@ -14,18 +14,53 @@ inputs:
+          echo "CCACHE_DIR=${{ inputs.current_cache_dir }}" >> $GITHUB_ENV
+          
+          # Configure settings for current cache
+          export CONF_PATH="${{ inputs.current_cache_dir }}/ccache.conf"
```
same

sublimator commented:
Oops

Will fix after gym unless you want to sort it yourself

sublimator committed:
refactor: address review comments properly

- remove redundant cache directory parameters from workflows
- use single ccache config file location at ~/.ccache/ccache.conf
- keep using ccache --set-config for dynamic settings
